### DISCUSSION TITLE
Implement a hardware accelerator in gem5

### DISCUSSION THREAD

**Discussion by fotuodunwu** [gem5/gem5#1363] [Q&A] [answered]
How to Implement a hardware accelerator in gem5? I have researched many papers and found similar variants of gem5, such as gem5-aladdin, gem5-salam, and gem5-accel. I am not sure what the differences are between these projects and gem5 itself. I have a rough understanding of how to create a new object based on SimObject in gem5, but I do not know how to model a hardware accelerator. Could someone provide a simple example? For instance, how to attach a very simple hardware module to the SoC, run the simulation, and compare the performance of implementing the same algorithm on the software side versus accelerating the algorithm with the hardware module. Or, more specifically, using 4 PEs to complete a 4x4 sparse matrix multiplication operation. If someone could provide an answer, I would be very grateful.

**Comment by ferry-hhh** [comment, marked as answer]
This simple DAXPY accelerator might be what you need: https://github.com/powerjg/gem5/tree/devel/accel

**Reply by fotuodunwu** [reply]
Thank you very much！
